src/reactors/PriorityOrderReactor.sol

```diff
@@ -56,17 +67,41 @@ contract PriorityOrderReactor is BaseReactor {
         );
     }
 
+    /// @notice update the priority order with the cosigner's data
+    /// @dev only called if the current block is before the auctionStartBlock signed by the user
+    /// @param orderHash the hash of the order
+    /// @param order the order to update
+    function _updateWithCosignerData(bytes32 orderHash, PriorityOrder memory order) internal pure {
+        /// return quickly if cosignerData is not set
+        if (order.cosignerData.auctionTargetBlock == 0) return;
+
+        if (order.cosignerData.auctionTargetBlock < order.auctionStartBlock) {
+            order.auctionStartBlock = order.cosignerData.auctionTargetBlock;
+        }
+        // validate cosigner signature
+        (bytes32 r, bytes32 s) = abi.decode(order.cosignature, (bytes32, bytes32));
```

Comment:
nit: may be nicer to have a small sig verification library now that we're doing this both here and in xv2